src/core/ucc_context.h

```diff
@@ -72,9 +81,27 @@ typedef struct ucc_context_config {
 ucc_status_t ucc_context_progress_register(ucc_context_t *ctx,
                                            ucc_context_progress_fn_t fn,
                                            void *progress_arg);
+
 void         ucc_context_progress_deregister(ucc_context_t *ctx,
                                              ucc_context_progress_fn_t fn,
                                              void *progress_arg);
+/* Performs address exchange between the processes group defined by OOB.
+   This function can be used either at context creation time
+   (if ctx is global) or at team creation time. The corresponding oob
+   arguments must be provided (c_oob for context and t_oob for team).
+   If used at context level the function is blocking, otherwise it
+   is non-blocking and can return UCC_INPROGRESS. In that case it
+   must be called until UCC_OK is returned.
+
+   The addresses are stored in the addr_storage data structure.
+
+   The addressing data of rank "i" (according to OOB) can be accessed
+   with UCC_ADDR_STORAGE_RANK_HEADER macro defined below.
+*/
+ucc_status_t ucc_core_addr_exchange(ucc_context_t          *context,
```

Comment:
I think it’s cleaner to make this function nonblocking only. And caller should implement blocking behavior if needed.